pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -195,6 +211,13 @@ public DataTable processQuery(ServerQueryRequest queryRequest, ExecutorService e
     long queryProcessingTime = queryProcessingTimer.getDurationMs();
     dataTable.getMetadata().put(DataTable.NUM_SEGMENTS_QUERIED, Integer.toString(numSegmentsQueried));
     dataTable.getMetadata().put(DataTable.TIME_USED_MS_METADATA_KEY, Long.toString(queryProcessingTime));
+
+    if (missingSegments > 0) {
+      dataTable.addException(QueryException.getException(QueryException.SEGMENTS_MISSING_ERROR,
```

Comment:
It may be worth adding a log message here with the name of the segments missing, Can help us debug

Reply:
Good point; will add it around line 138

Reply:
Done.